src/Symfony/Component/HttpKernel/RebootableInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel;
+
+/**
+ * Rebotable extends the Kernel to allow it to reboot using a temporary cache directory.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+interface RebootableInterface
+{
+    /**
+     * Reboots a kernel.
+     *
+     * The getCacheDir() method of a rebootable kernel should not be called
+     * while building the container. Use the %kernel.cache_dir% parameter instead.
+     *
+     * @param string|null $warmupDir
+     */
+    public function reboot($warmupDir);
```

Comment:
+     * @param string|null $warmupDir pass `null` to reboot to the regular cache dir